networking/src/Mockable/Production.hs

```diff
@@ -1,3 +1,5 @@
+-- `MonadMask Production` is missing `generalBracket`.
+{-# OPTIONS_GHC -Wno-missing-methods    #-}
```

Comment:
That seems like an important thing to fix to me. The new `exceptions` library defines everything in terms of `generalBracket`, so it seems to me like this should cause breakage.

Reply:
Right! If someone who is more familiar with the code knows how the proper generalBracket implementation has to look can just provide it, that would be helpful :-)

Reply:
`Production` is a newtype around `IO` so we can just use `GeneralizedNewtypeDeriving` to derive the instance.